Mathlib/Data/Seq/WSeq.lean

```diff
@@ -1329,7 +1331,12 @@ theorem toList'_map (l : List α) (s : WSeq α) :
 
 @[simp]
 theorem toList_cons (a : α) (s) : toList (cons a s) = (List.cons a <$> toList s).think :=
-  destruct_eq_think <| by unfold toList; simp; rw [toList'_map]; simp; rfl
+  destruct_eq_think <| by
+    unfold toList
+    simp only [toList'_cons, Computation.destruct_think, Sum.inr.injEq]
+    rw [toList'_map]
+    simp only [List.reverse_cons, List.reverse_nil, List.nil_append, List.singleton_append]
+    rfl
```

Comment:
This looks like a symptom of the simp bug that was fixed by letting simp not resynthesize everything anymore. Hence I'm pretty sure something like this will work:
```suggestion
  destruct_eq_think <| by unfold toList; simp [toList'_map]; rfl
```

Reply:
Literally what you suggest gives
```lean
type mismatch
  HEq.rfl
has type
  HEq ?m.129498 ?m.129498 : Prop
but is expected to have type
  Computation.destruct (toList (cons a s)) = Sum.inr (List.cons a <$> toList s) : Prop
```

I could not find a better proof...